### PR TITLE
Apply signaling component -- rebase

### PR DESCRIPTION
**Pull request by ActoryOu**
*Issue #, if available:*

*What was changed?*

- Add [Signaling component](https://github.com/awslabs/amazon-kinesis-video-streams-signaling) and [coreJSON](https://github.com/FreeRTOS/coreJSON) as external projects, and update Signaling code to call APIs of component.

*Why was it changed?*

- To apply Signaling re-factoring component into dev branch. [Signaling component](https://github.com/awslabs/amazon-kinesis-video-streams-signaling) is separated out as independent library. And coreJSON is used in [Signaling component](https://github.com/awslabs/amazon-kinesis-video-streams-signaling).

*How was it changed?*

- Add [Signaling component](https://github.com/awslabs/amazon-kinesis-video-streams-signaling) and [coreJSON](https://github.com/FreeRTOS/coreJSON) as external projects, import it via ExternalProject_Add, and update Signaling code to call APIs of component.

*What testing was done for the changes?*

- Verified master demo on Ubuntu
- Verified webrtc_client_test and pass all test cases.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CMake/Dependencies/libcorejson-add-cmakelists.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (moninom1) [inline CMake/Dependencies/libcorejson-add-cmakelists.patch:1]: Is this added by mistake?

Comment (ActoryOu) [reply CMake/Dependencies/libcorejson-add-cmakelists.patch:1]: No, that's not a mistake. There is no CMakeLists.txt in coreJSON repo. And I add that file into that repo in compile time by applying this patch.

File: src/source/Signaling/LwsApiCalls.c [hunk 2]
```diff
@@ -885,44 +980,47 @@ STATUS createChannelLws(PSignalingClient pSignalingClient, UINT64 time)
...
+    SignalingAwsRegion_t awsRegion;
+    CreateSignalingChannelRequestInfo_t createSignalingChannelRequestInfo;
+    SignalingChannelArn_t channelArn;
+    SignalingTag_t tags[MAX_TAG_COUNT];
 
     CHK(pSignalingClient != NULL, STATUS_NULL_ARG);
-
+    CHK(pSignalingClient->pChannelInfo->tagCount <= MAX_TAG_COUNT, STATUS_INVALID_ARG);
+
+    // Prepare AWS region
```
[the remark this answers is not in the document]
Comment (ActoryOu) [reply src/source/Signaling/LwsApiCalls.c:996]: We should keep this aligned in this repo. So let's keep it as it is.

**Review by shubnil — APPROVED** [review]

**Review by hassanctech — APPROVED** [review]
nice!  very excited about this rework!

**Pull request merged**